src/ex2/script.js

```diff
@@ -0,0 +1,146 @@
+class Main {
+    constructor() { }
+
+    init() {
+        //add click event listener to add button
+        addBtn.addEventListener("click", () => { this.addTodoAndRender() })
+
+        // add a task when user press Enter key
+        taskInput.addEventListener("keypress", (e) => { if (e.key === 'Enter' && taskInput.value) this.addTodoAndRender() });
+
+        //Clear the tasks list
+        clearAllBtn.addEventListener("click", () => {
+            itemManager.clearArr();
+            taskList.innerHTML = ''; //Clear the list of tasks element
+            this.updateForm();
+        })
+
+        //sort the tasks list add them to html list element and save them to the LocalStorage
+        sortAllBtn.addEventListener("click", () => {
+            itemManager.sortArr();
+            taskList.innerHTML = '';
+            this.renderAllTasks();
+        })
+
+        //waiting for user input before enable the add button
+        taskInput.addEventListener('keyup', () => addBtn.disabled = !taskInput.value)
+
+        if (itemManager.taskArr.length != 0) this.renderAllTasks();
+        else placeHolder.classList.remove("hidden");
+    }
+
+    //function to add task and update the list and the localStorage
+    addLiTask(taskValue) {
+        taskList.appendChild(this.createDOMTask(taskValue));
+        taskInput.value = ''; //clear the input after added
+        addBtn.disabled = true;
+        this.updateForm();
+    }
+
+    // function to remove from DOM task that have been clicked
+    removeLiTask(li) {
+        taskList.removeChild(li);
+        this.updateForm();
+    }
+
+    //update the button and the placeHolder if necessary
+    updateForm() {
+        pedingTaskText.innerHTML = "You have " + itemManager.taskArr.length + " pending tasks"
+        if (itemManager.taskArr.length === 0) { //if there are no task any more disable the clear and sort button
+            clearAllBtn.disabled = true;
+            sortAllBtn.disabled = true;
+            placeHolder.classList.remove("hidden");
+        }
+        else {
+            clearAllBtn.disabled = false;
+            sortAllBtn.disabled = false;
+            placeHolder.classList.add("hidden");
+        }
+    }
+
+    //function to insert the all tasks from the localStorage to the html list element
+    renderAllTasks() {
+        itemManager.taskArr.forEach((taskValue) => {
+            taskList.appendChild(this.createDOMTask(taskValue));
+        });
+        this.updateForm();
+    }
+
+    //Function to create the li element and its childs and assign a click event to them
+    createDOMTask(taskValue) {
```

Comment:
would have liked to see these separated out into their own services, this file is very long and does a lot of different things, makes it harder to read and understand the code